### PR TITLE
feat: Update “download datasets” copy URL (#5589)

### PR DESCRIPTION
**Pull request by frano-m**
## Reason for Change

- #5589
- https://github.com/chanzuckerberg/single-cell-data-portal/issues/5586

## Changes
- Modified download datasets copy URL.
- Modified download datasets copy caption text as per [mocks](https://www.figma.com/file/jEv0ZWQd0Q53aFl9nNRbBy/Portal?node-id=192%3A51303&mode=dev) to include a link to the schema and Discover API.
- Schema URL is updated to https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.1.0/schema.md as per [slack thread](https://clevercanary.slack.com/archives/C02CF7MQPGV/p1694028293166609?thread_ts=1694026581.496909&cid=C02CF7MQPGV).

### CODE REVIEW

File: frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/index.tsx
Comment (danieljhegeman) [inline frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/index.tsx:1]: @frano-m can you give me context on what these js changes are doing?

**Review by danieljhegeman — COMMENTED** [review]
@frano-m I'm not seeing where the changes are being introduced that replace a presigned url download link with a `https://datasets.cellxgene.cziscience.com/<dataset_version_id>.<ext>` link, per #5589

**Comment by danieljhegeman** [comment]
My cursory assessment is that the FE is still calling [this endpoint](https://github.com/chanzuckerberg/single-cell-data-portal/blob/505d9d362f7cec5851eb4793b117f85e95933188/backend/portal/api/portal-api.yml#L549) which (still) generates a presigned url.

**Review by danieljhegeman — CHANGES_REQUESTED** [review]
See comments

**Comment by MillenniumFalconMechanic** [comment]
> My cursory assessment is that the FE is still calling [this endpoint](https://github.com/chanzuckerberg/single-cell-data-portal/blob/505d9d362f7cec5851eb4793b117f85e95933188/backend/portal/api/portal-api.yml#L549) which (still) generates a presigned url.
> 
> Copying from #5589:
> 
> > Currently, a request is sent to create and retrieve a pre-signed URL. The generated pre-signed URL is no longer required; the URL is to be assembled by the frontend from the dataset version ID.

Hi @danieljhegeman, we changed the approach on how the permanent URL should be generated but it looks like I didn't update the "need" section of the corresponding ticket. I have just made that update. 

The update to call the new endpoint is [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5835/files#diff-fad645b29fca0b75699eff24024074dee4b587f47fbbc95355b6e0558e3137beR74).

The existing POST endpoint will be cleaned up in [this ticket](https://github.com/chanzuckerberg/single-cell-data-portal/issues/5697).

**Review by tihuan — APPROVED** [review]
LGTM 😄 👍 !

File: frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/index.tsx [hunk 1]
```diff
@@ -69,15 +71,15 @@ const Content: FC<Props> = ({ onClose, name, dataAssets }) => {
         const result = await (
           await fetch(`${API_URL}${url}`, {
             ...DEFAULT_FETCH_OPTIONS,
-            method: "POST",
+            method: "GET",
```
[the remark this answers is not in the document]
Comment (danieljhegeman) [reply frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/index.tsx:74]: I don't think so... I think we want to move away from the presigned s3 url and toward using the Discover API permalink for a given Dataset version.

**Comment by danieljhegeman** [comment]
> The update to call the new endpoint is [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5835/files#diff-fad645b29fca0b75699eff24024074dee4b587f47fbbc95355b6e0558e3137beR74).

This is not a new endpoint, as you allude to in a subsequent comment. It is the same endpoint, just a different method, right?

**Comment by danieljhegeman** [comment]
> Hi @danieljhegeman, we changed the approach on how the permanent URL should be generated but it looks like I didn't update the "need" section of the corresponding ticket. I have just made that update.

Can you say a bit more about this?

**Comment by MillenniumFalconMechanic** [comment]
> > The update to call the new endpoint is [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5835/files#diff-fad645b29fca0b75699eff24024074dee4b587f47fbbc95355b6e0558e3137beR74).
> 
> This is not a new endpoint, as you allude to in a subsequent comment. It is the same endpoint, just a different method, right?

Ah, I see! Yes, we added a `GET` method to `/datasets/datasetId/asset/assetId` in [this PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5794), specifically [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/portal/api/portal-api.yml#L550). 

The existing `POST` method (e.g. [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/portal/api/portal-api.yml#L584)) will be removed with [Remove Data Portal pre-signed asset URL capability](https://github.com/chanzuckerberg/single-cell-data-portal/issues/5697) once this PR lands.

> Can you say a bit more about this?

Originally the approach was to generate the link to the permanent URL on the FE and deprecate the existing `POST` method. During our weekly call on September 13, (@frano-m and) I raised the issue that we need to display a file size in the download modal which is currently provided to the FE via the soon-to-be-deprecated `POST` endpoint. @ebezzi suggested we create a new `GET` method to return both the permanent URL (as we could then re-use the logic used by the curator API) and the file size.

@danieljhegeman, let me know if this isn't clear or if you have any more q's!

**Comment by danieljhegeman** [comment]
> > > The update to call the new endpoint is [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5835/files#diff-fad645b29fca0b75699eff24024074dee4b587f47fbbc95355b6e0558e3137beR74).
> > 
> > 
> > This is not a new endpoint, as you allude to in a subsequent comment. It is the same endpoint, just a different method, right?
> 
> Ah, I see! Yes, we added a `GET` method to `/datasets/datasetId/asset/assetId` in [this PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/5794), specifically [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/portal/api/portal-api.yml#L550).
> 
> The existing `POST` method (e.g. [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/portal/api/portal-api.yml#L584)) will be removed with [Remove Data Portal pre-signed asset URL capability](https://github.com/chanzuckerberg/single-cell-data-portal/issues/5697) once this PR lands.
> 
> > Can you say a bit more about this?
> 
> Originally the approach was to generate the link to the permanent URL on the FE and deprecate the existing `POST` method. During our weekly call on September 13, (@frano-m and) I raised the issue that we need to display a file size in the download modal which is currently provided to the FE via the soon-to-be-deprecated `POST` endpoint. @ebezzi suggested we create a new `GET` method to return both the permanent URL (as we could then re-use the logic used by the curator API) and the file size.
> 
> @danieljhegeman, let me know if this isn't clear or if you have any more q's!

Thanks @MillenniumFalconMechanic . I think this work just came in / is coming in more piecemeal than I might've expected, but all the same in terms of the end goal. Thanks for confirming!

**Review by danieljhegeman — APPROVED** [review]
📈

**Review by tihuan — APPROVED** [review]

**Pull request merged**